LanguageFeatures/Patterns/scope_A05_t01.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The variables defined by a pattern and its subpatterns (its
+/// pattern variable set, defined above), are introduced into a scope based on
+/// where the pattern appears:
+/// ...
+/// Switch statement, switch expression, if-case statement, if-case-element:
+/// Each guardedPattern introduces a new case scope which is where the variables
+/// defined by that case's pattern are bound.
+/// ...
+/// It is a compile-time error for a guard to contain an assignment to a
+/// variable defined in the case that owns that guard.
+///
+/// @description Checks that it is a compile-time error for a guard to contain
+/// an assignment to a variable defined in the case that owns that guard.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+void test1() {
+  switch (42) {
+    case var v when (v = 1) > 0:
+//                   ^
+// [analyzer] unspecified
+// [cfe] unspecified
+      print(v);
+      break;
+    default:
+  }
+}
+
+void test2() {
+  var z = switch (42) {
+    final int v when (v = 1) => 0 => v,
```

Comment:
I think this construct is an error (essentially a syntax error), because of the ambiguity around `=>`. In order to make sure the error is the relevant one we'd probably have to use `()` around the function literal.

Reply:
Yes, it's a typo. There should be `>=`, not a `=>`